security-analysis-ws/src/main/java/eu/itesla_project/security/rest/api/impl/SecurityAnalysisServiceImpl.java

```diff
@@ -179,4 +183,140 @@ private FilePart getFilePart(List<InputPart> parts) throws IOException {
         return null;
     }
 
+    @Override
+    public Response actionSimulator(MultipartFormDataInput form) {
+        Objects.requireNonNull(form);
+        try {
+            Map<String, List<InputPart>> formParts = form.getFormDataMap();
+            Format format = getFormat(formParts);
+
+            if (format == null) {
+                return Response.status(Status.BAD_REQUEST).entity("Missing required format parameter").build();
+            }
+
+            FilePart caseFile = formParts.get("case-file") != null ? getFilePart(formParts.get("case-file")) : null;
+            if (caseFile == null) {
+                return Response.status(Status.BAD_REQUEST).entity("Missing required case-file parameter").build();
+            }
+
+            FilePart dslFile = formParts.get("dsl-file") != null ? getFilePart(formParts.get("dsl-file")) : null;
+            if (dslFile == null) {
+                return Response.status(Status.BAD_REQUEST).entity("Missing required dsl-file parameter").build();
+            }
+
+            Network network = Importers.loadNetwork(caseFile.getFilename(), caseFile.getInputStream());
+
+            boolean verbose = false;
+
+            List<String> contingencies = Collections.emptyList();
+
+            StringWriter writer = new StringWriter();
+            IOUtils.copy(dslFile.getInputStream(), writer, StandardCharsets.UTF_8);
+            String dslAsString = writer.toString();
+
+            // load actions from Groovy DSL
+            ActionDb actionDb = new ActionDslLoader(dslAsString).load(network,
+                    new ActionDslLoaderObserver(new PrintStream(new ByteArrayOutputStream()), verbose));
+
+            if (contingencies.isEmpty()) {
+                contingencies = actionDb.getContingencies().stream().map(Contingency::getId)
+                        .collect(Collectors.toList());
+            }
+
+            LoadFlowActionSimulatorConfig config = LoadFlowActionSimulatorConfig.load();
+
+            List<LoadFlowActionSimulatorObserver> observers = new ArrayList<>();
+            AbstractSecurityAnalysisResultBuilderImpl loadFlowActionSimulatorObserver = new AbstractSecurityAnalysisResultBuilderImpl();
+            observers.add(loadFlowActionSimulatorObserver);
+
+            // action simulator
+            ActionSimulator actionSimulator = new LoadFlowActionSimulator(network, new LocalComputationManager(), config,
+                    observers);
+
+            // start simulator
+            actionSimulator.start(actionDb, contingencies);
+
+            SecurityAnalysisResult securityAnalysisResult = loadFlowActionSimulatorObserver.getResult();
```

Comment:
Maybe we could directly build the "StreamingOuput" in the observer implementation, and get it here.
It's not very different, do as you prefer.